platform/darwin/src/MGLStyleValue.mm

```diff
@@ -1,8 +1,130 @@
+
 const MGLStyleFunctionOption MGLStyleFunctionOptionInterpolationBase = @"MGLStyleFunctionOptionInterpolationBase";
 const MGLStyleFunctionOption MGLStyleFunctionOptionDefaultValue = @"MGLStyleFunctionOptionDefaultValue";
 
+id MGLJSONObjectFromMBGLValue(const mbgl::style::expression::Value &value) {
+    return value.match([](const mbgl::NullValue) -> id {
+        return [NSNull null];
+    }, [](const bool value) {
+        return @(value);
+    }, [](const float value) {
+        return @(value);
+    }, [](const int64_t value) {
+        return @(value);
+    }, [](const double value) {
+        return @(value);
+    }, [](const std::string &value) {
+        return @(value.c_str());
+    }, [](const mbgl::Color &value) {
+        return [MGLColor mgl_colorWithColor:value];
+    }, [](const mbgl::style::Position &value) {
+        std::array<float, 3> spherical = value.getSpherical();
+        MGLSphericalPosition position = MGLSphericalPositionMake(spherical[0], spherical[1], spherical[2]);
+        return [NSValue valueWithMGLSphericalPosition:position];
+    }, [&](const std::vector<mbgl::style::expression::Value> &vector) {
+        NSMutableArray *array = [NSMutableArray arrayWithCapacity:vector.size()];
+        for (auto value : vector) {
+            [array addObject:MGLJSONObjectFromMBGLValue(value)];
+        }
+        return @[@"literal", array];
+    }, [&](const std::unordered_map<std::string, mbgl::style::expression::Value> &map) {
+        NSMutableDictionary *dictionary = [NSMutableDictionary dictionaryWithCapacity:map.size()];
+        for (auto &item : map) {
+            dictionary[@(item.first.c_str())] = MGLJSONObjectFromMBGLValue(item.second);
+        }
+        return @[@"literal", dictionary];
+    }, [](const auto &) -> id {
+        return nil;
+    });
+}
+
+id MGLJSONObjectFromMBGLExpression(const mbgl::style::expression::Expression &mbglExpression) {
+    using namespace mbgl::style::expression;
+    if (auto literalExpression = dynamic_cast<const Literal *>(&mbglExpression)) {
+        auto result = literalExpression->evaluate({ nullptr });
+        return result ? MGLJSONObjectFromMBGLValue(*result) : nil;
+    }
+    if (auto assertExpression = dynamic_cast<const ArrayAssertion *>(&mbglExpression)) {
+#warning Implement array assertion expressions.
+        NSMutableArray *inputs = [NSMutableArray array];
+        assertExpression->eachChild([&](const Expression &child) {
+            [inputs addObject:MGLJSONObjectFromMBGLExpression(child)];
+        });
+        return @[@"literal", inputs.lastObject];
+    }
+    if (auto assertExpression = dynamic_cast<const Assertion *>(&mbglExpression)) {
+#warning Implement assertion expressions.
+        NSMutableArray *inputs = [NSMutableArray array];
+        assertExpression->eachChild([&](const Expression &child) {
+            [inputs addObject:MGLJSONObjectFromMBGLExpression(child)];
+        });
+//        type::Type type = assertExpression->getType();
```

Comment:
Nit: If this code is not used could you please remove it.

Reply:
* [ ] File tail-work ticket about implementing type assertions
* [ ] Remove commented out type assertion code